include/QtNodes/internal/NodeDelegateModel.hpp

```diff
@@ -31,6 +31,19 @@ struct NodeValidationState
     QString _stateMessage{""};
 };
 
+/**
+ * Describes the node status, depending on its current situation
+ */
+enum class NodeProcessingStatus : int {
+    NoStatus = 0,   ///
+    Updated = 1,    ///
+    Processing = 2, ///
+    Pending = 3,    ///
+    Empty = 4,      ///
+    Failed = 5,     ///
+    Partial = 6,    ///
+};
```

Comment:
Maybe some use-cases or a short comment could help others?
What is the difference between `NoStatus` and `Empty`?
Between `Processing` and `Partial`?